src/components/__tests__/Tasks.test.js

```diff
@@ -0,0 +1,139 @@
+import { render, screen, waitFor } from "@testing-library/react"
```

Comment:
These tests all look really good. Might be able to pull some of the test setup into a help function but other than that, excellent work. If you did the state updates i mentioned in the api file, then you won't need the mocking of getItems and can just test against the displayed items instead (as they will be updated in real time i guess?)